ui/src/app/cohort-search/cohort-page/cohort-page.component.tsx

```diff
@@ -120,16 +117,22 @@ export const CohortPage = fp.flow(withCurrentWorkspace(), withCurrentCohortSearc
         };
         localStorage.setItem(LOCAL_STORAGE_KEY_COHORT_SEARCH_REQUEST, JSON.stringify(localStorageCohort));
       }));
-      this.props.setShowWarningModal(this.showWarningModal);
+      this.unblock = this.props.history.block(async() => {
```

Comment:
Tried to adapt a suggestion from an issue on react-router's github page: https://github.com/ReactTraining/react-router/issues/4635#issuecomment-297828995

Also tried a similar method seen here: https://github.com/ReactTraining/history/blob/master/docs/blocking-transitions.md

Never could get `history.block()` to fire with either.